### ISSUE TITLE
Running OWLViTv2 on non-square images provides upwards shifted bounding boxes

### ISSUE THREAD

**Issue by Thomas-Malchers** [NielsRogge/Transformers-Tutorials#356]
Hey,
I am trying to run your owlvit v2 notebook on my local machine. whenever i am using an image that is not of squared shape, such as the cat example that you also use, the resulting bounding boxes are slightly shifted upwards. I tried multiple images, filetypes, etc. and the problem still persists. Is there some parameter i am missing? The problem also does not seem to be in the plotting itself as the coordinates of the bounding boxes are wrong already.

![test](https://github.com/NielsRogge/Transformers-Tutorials/assets/70331927/bcdc08dd-bc2e-4f3e-8848-cb59a80c49ff)

![test_2](https://github.com/NielsRogge/Transformers-Tutorials/assets/70331927/19de0827-de13-43b7-b58d-859f66d48d54)

**Comment by NielsRogge** [comment]
Thanks for reporting, will look into it.

In the meantime could you try using `OwlViTProcessor` instead of `Owlv2Processor` and see whether you get better results?

**Comment by Thomas-Malchers** [comment]
Thanks for the quick reply. In the meantime, i had a look at the OwlViT v1 Processor and when interchanging the processor in this notebook: https://github.com/huggingface/notebooks/blob/main/examples/zeroshot_object_detection_with_owlvit.ipynb with the updated v2 models/ processors it seems to work. I assume that the direct access on the results instead of calling `results = processor.post_process_object_detection(outputs=outputs, target_sizes=target_sizes, threshold=0.2)` makes the difference.

**Comment by NielsRogge** [comment]
Yeah I've run the [original Colab](https://github.com/google-research/scenic/blob/main/scenic/projects/owl_vit/notebooks/OWL_ViT_minimal_example.ipynb) by the authors on the cats image, when visualizing the bounding boxes they visualize them on the padded image, not on the original image.

![image](https://github.com/NielsRogge/Transformers-Tutorials/assets/48327001/aafd1a62-1de2-42e9-9e07-5b1fbddc3462)

So we need to visualize them on the padded image rather than the original one. Will look more into this over the weekend
